src/odemis/driver/pigcs.py

```diff
@@ -2396,24 +2389,22 @@ def stopMotion(self):
 
     def startReferencing(self, axis):
         """
-        Start a referencing move. Use isMoving() or isReferenced() to know if
-        the move is over. Position will change, as well as absolute positions.
+        Start a referencing move. Position will change, as well as absolute positions.
         axis (str)
         """
         self._acquireAxis(axis)
 
-        # Note: setting position only works if ron is disabled. It's possible
-        # also to indirectly set it after referencing, but then it will conflict
+        # Note: Setting position only works if ron is disabled when not referencable.
+        # It's possible also to indirectly set it after referencing, but then it will conflict
         # with TMN/TMX and some correct moves will fail.
-        # So referencing could look like:
-        # ron 1 1
-        # frf -> go home and now know the position officially
-        # orig_pos = pos?
 
         if self._hasRefSwitch[axis]:
             self.SetReferenceMode(axis, True)
             self.ReferenceToSwitch(axis)
+            # FIXME: If referencing fails, this flag might be True though it should be not.
```

Comment:
That's true, but as it's also compared with IsReferenced(), I don't think it's too big an issue. At least, as long as the startServo() (and so RON) is not called, that should be detected anyway, and the `_referenced` flag reset to False.